src/seedu/addressbook/commands/Command.java

```diff
@@ -37,6 +37,16 @@
     protected List<? extends Assessment> relevantAssessments;
     protected List<? extends ReadOnlyExam> relevantExams;
     private int targetIndex = -1;
+    private int targetExamIndex = -1;
```

Comment:
Feels a bit bad to add targetExamIndex attribute to all commands. Maybe have a another subclass of commands that exam commands inherit from instead?

Reply:
i need to refer to both lists in 2 of the commands though

Reply:
So what I meant was create another class ExamCommand or something that adds in this attribute, rather than having all command object to contain this attribute

Reply:
![image](https://user-images.githubusercontent.com/34745754/47265408-20458500-d55a-11e8-9c30-0d877259af79.png)

and 

![image](https://user-images.githubusercontent.com/34745754/47265412-32272800-d55a-11e8-9488-c2131fefd5e3.png).
targetIndex is added to all commands anyway, even to list command, I dont see why its bad to have it in command

Reply:
Copying bad code doesn't make it a good code.
Target index is even more general than your exam index. Around 50% of commands have a target index, while only your subset have ExamIndex, that's why it was passable, albeit bad.
Imagine if I had one for Account index, one for privilege index, then everyone else have one index for one feature, it will bloat it up unnecessarily and reduce cohesion.